parmed/amber/_amberparm.py

```diff
@@ -2110,16 +2103,14 @@ def _cleanup_dihedrals_with_periodicity_zero(self):
         terms that have pn==0, and simply add another dihedral with pn=1 and k=0 to ensure that
         pmemd will always get that exception correct
         """
-        dt = None
         new_dihedrals = []
         for dih in self.dihedrals:
             if dih.ignore_end or dih.type.per != 0:
                 continue
             # If we got here, ignore_end must be False and out periodicity must be 0. So add
             # another dihedral
-            if dt is None:
```

Comment:
This was the problem.  It ran the assumption that all 1-4 scaling factors were the same, which they're not.  This was causing 1-4 scaling factors to be inadvertently changed (which this latest fix avoids).